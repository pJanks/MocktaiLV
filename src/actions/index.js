export const addUser = (user) => ({
  type: 'ADD_USER',
  user
})

export const addDrinks = (drinks) => ({
  type: 'LOAD_NONALCOHOLIC_DRINKS',
  drinks
})

export const addFavoriteDrink = (favoriteDrink) => ({
  type: 'ADD_FAVORITE_DRINK',
  favoriteDrink
})

export const getDrink = (drink) => ({
  type: 'GET_DRINK',
  drink
})
