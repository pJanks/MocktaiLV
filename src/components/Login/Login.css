.login {
  display: flex;
  /* flex-direction: column; */
}

.login-content {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin: 20px;
}

.login-form {
  align-items: center;
  border: 2px solid #000;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: 8px;
  width: 40%;
}

input {
  color: #000;
  height: 40px;
  margin: 10px;
  padding-left: 4px;
  width: 50%;
}

button {
  margin: 10px;
  padding: 8px;
  text-decoration: none;
  color: #000;
}

.link {
  color: #000;
  margin: 10px;
  padding: 8px;
  text-decoration: none;
}

h4 {
  width: 60%;
}
