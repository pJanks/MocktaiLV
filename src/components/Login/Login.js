import React, { Component } from 'react';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { addUser } from '../../actions'
import AllMocktails from '../AllMocktails/AllMocktails'

export class Login extends Component {
  constructor() {
    super()
    this.state = {
      username: '',
      password: '',
      error: false,
      userVerified: false,
      isGuest: false
    }
  }

  handleChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  checkLocalStorage = (name, password) => {
    let keys = Object.keys(localStorage)
    let user = keys.find(key => {
      return (JSON.parse(localStorage[key]).name === name && JSON.parse(localStorage[key]).password === password)
    })
    return user
  }

  handleLoginButtonClick = async (event, username, password) => {
    event.preventDefault();
    let validatedUserStorage = this.checkLocalStorage(username, password)
    if(username && password && validatedUserStorage) {
      await this.setState({ userVerified: true })
      this.props.addUserToStore({
        name: this.state.username,
        userVerified: true
      })
    } else {
      window.alert('You Must Sign Up')
    }
  }

  handleGuestButtonClick = (event) => {
    event.preventDefault()
    this.setState({ isGuest: true })
  }

  render() {
    if(this.state.userVerified || this.state.isGuest) {
      return <Redirect to= '/AllMocktails' />
    } else {
      return (
        <form>
          <input
            type="text"
            name="username"
            placeholder="Username"
            onChange={this.handleChange}
          />
          <input
            type="password"
            name="password"
            placeholder="Password"
            onChange={this.handleChange}
          />
          <button onClick={(event) => this.handleLoginButtonClick(event, this.state.username, this.state.password)}>Login</button>
          <button onClick={(event) => this.handleGuestButtonClick(event)}>Continue as Guest</button>
          <button onClick={(event) => this.handleSignUpButtonClick(event)}><Link to='/SignUp'>Sign Up</Link></button>
        </form>
      )
    }
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addUserToStore: (user) => dispatch(addUser(user))
})

export default connect(null, mapDispatchToProps)(Login);
