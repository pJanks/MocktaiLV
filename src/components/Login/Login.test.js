import React from 'react';
import { Login, mapDispatchToProps } from './Login';
import { shallow } from 'enzyme';
import { addUserToStore } from '../../actions';

describe('Login', () => {
  let wrapper, mockUser;

  beforeEach(() => {
    wrapper = shallow(<Login />)
    mockUser = {
      name: 'Greg',
      userVerified: true
    }
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  it('should track user input entry', () => {
    const mockUserEvent = { target: { name: 'user', value: 'user1' } };
    const mockPassEvent = { target: { name: 'password', value: 'passTest' } };

    wrapper.instance().handleChange = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('[name="username"]').simulate('change', mockUserEvent);
    wrapper.find('[name="password"]').simulate('change', mockPassEvent);

    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockUserEvent);
    expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockPassEvent);        });
});
