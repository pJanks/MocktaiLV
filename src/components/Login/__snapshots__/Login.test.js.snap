// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<main
  className="login"
>
  <div
    className="login-content"
  >
    <form
      className="login-form"
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="text"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
      <button
        onClick={[Function]}
      >
        Login
      </button>
      <button
        onClick={[Function]}
      >
        Continue as Guest
      </button>
      <button>
        <Link
          className="link"
          to="/SignUp"
        >
          Sign Up
        </Link>
      </button>
    </form>
    <br />
    <h4>
      Find drinks to enjoy while your enjoying yourself in Sin City without the intoxicating effects of alcohol (if you so chose). Otherwise feel free to browse our alcoholc drinks in the 'Off the Wagon' section!
    </h4>
  </div>
</main>
`;
